engine/lib/import_thir.ml

```diff
@@ -1091,7 +1095,36 @@ end) : EXPR = struct
     let arm_pat = c_pat arm.pattern in
     let body = c_expr arm.body in
     let span = Span.of_thir arm.span in
-    { arm = { arm_pat; body }; span }
+    let guard =
+      Option.map
+        ~f:(fun (e : Thir.decorated_for__expr_kind) ->
+          match e.contents with
+          | Let { expr; pat } ->
+              {
+                guard =
+                  IfLet
+                    {
+                      lhs = c_pat pat;
+                      rhs = c_expr expr;
+                      witness = W.match_guard;
+                    };
+                span = Span.of_thir e.span;
+              }
+          | _ ->
+              {
+                guard =
+                  IfLet
+                    {
+                      lhs =
+                        { p = PConstant { lit = Bool true }; span; typ = TBool };
+                      rhs = c_expr e;
+                      witness = W.match_guard;
+                    };
+                span = Span.of_thir e.span;
+              })
+        arm.guard
```

Comment:
Can you put the match in a `let` binding? e.g. a `let guard = match ... with ... in {guard; span = Span.of_thir ...}`?